sys/include/stdio_nimble.h

```diff
@@ -25,6 +21,137 @@
  *               write buffer since only 'stdio_write' uses this mutex. Characters will be written
  *               in FIFO mode. Characters that do not fit in the buffer will be dropped.
  *
+ * This module uses NimBLE for stdio. The bluetooth characteristic for
+ * stdin is writable and the characteristic for stdout uses the indicate
+ * mechanism to publish the system's output to a connected device. Data will be
+ * sent out asynchronously via callout functions.
+ *
+ * To use this module, add
+ * ```
+ * USEMODULE += stdio_nimble
+ * ```
+ * to your makefile.
+ *
+ * You can change the default buffer sizes by adding
+ * ```
+ * CFLAGS += -DCONFIG_STDIO_NIMBLE_STDIN_BUFSIZE=1024
+ * CFLAGS += -DCONFIG_STDIO_NIMBLE_STDOUT_BUFSIZE=2048
+ * ```
+ * to your makefile.
+ *
+ * **NOTE:** These values must be a power of two!
+ *
+ * By default, stdin and stdout buffers are cleared on a connect event. To keep the
+ * content add the following to your makefile:
+ * ```
+ * CFLAGS += -DCONFIG_STDIO_NIMBLE_CLEAR_BUFFER_ON_CONNECT=0
+ * ```
+ *
+ * For automatic bluetooth advertising a module is provided: *nimble_autoadv*.
+ * It will take care of enabling advertising on disconnect events and
+ * disabling advertising on connect events. It can be enabled by adding
+ * ```
+ * USEMODULE += nimble_autoadv
+ * ```
+ * to your makefile.
+ *
+ * The advertised device name can then optionally be configured with
+ * ```
+ * CFLAGS += -DCONFIG_NIMBLE_AUTOADV_DEVICE_NAME='"Riot OS device"'
+ * ```
+ * Otherwise the device will appear as "*RIOT OS device*".
+ *
+ * ## Instructions to connect to the bluetooth shell via ble-serial
+ *
+ * ### Configure and compile shell app
+ * This example is for the `nrf52840dongle` target in `tests/sys/shell`, other
+ * targets that are supported by NimBLE can be used similarly.
+ *
+ * Add following to Makefile:
+ * ```
+ * BOARD = nrf52840dongle
+ * USEMODULE += nimble_autoadv
+ * USEMODULE += stdio_nimble
+ * ```
+ *
+ * **NOTE:** You can also have a look at `tests/sys/shell_ble`, which is already
+ * set up for NimBLE.
+ *
+ * ### Flash
+ *
+ * ```
+ * $ make -C tests/sys/shell -j clean all flash
+ * ```
+ *
+ * ### Install the ble-serial tool
+ *
+ * ```
+ * $ pip install ble-serial
+ * ```
+ *
+ * ### Scan for your device
+ * To connect to the device, you have to find out the BLE address. For this
+ * example, the device name is `Riot OS device`. For the `tests/sys/shell_ble`
+ * test, the device name is `tests/sys/shell_ble`.
+ *   When you get `ble-scan: command not found` you can also run
+ *   `python -m ble_serial.scan` instead.
+ * ```
+ * $ ble-scan
+ * Started BLE scan
+ *
+ * 6BE8174C-A0F8-4479-AFA6-9828372CAFE9 (RSSI=-40): Riot OS device
```

Comment:
For me, `ble-scan` (from `ble-serial 3.0.0`) shows MAC Adresses instead of UUIDs, e.g. `F1:5C:B2:0F:07:1F (rssi=-45): Riot OS device`. Not sure if this is only my local environment, but if not, I would change this example here.

Reply:
That depends on the operating system. MacOS X shows the long form, Linux and Windows show MAC addresses.
Perhaps that would be worth mentioning 🤔 

I'll change the reference in `tests/sys/shell_ble` as well.